ondevice_learning.qmd

```diff
@@ -1,61 +1,630 @@
+
+![The cycle of federated learning from personal phones to the full-scale model.(Credit: [Google Research](https://ai.googleblog.com/2017/04/federated-learning-collaborative.html))](images/ondevice_intro.png)
 
-  - Importance in Embedded AI
-  - Why is On-device Learning Needed
 
 ## Advantages and Limitations
 
-Explanation: Understanding the pros and cons of on-device learning helps to identify the scenarios where it is most effective and the challenges that need to be addressed.
+On-Device Learning provides a number of advantages over traditional cloud-based machine learning. By keeping data and models on the device, it eliminates the need for costly data transmission and addresses privacy concerns. This allows for more personalized, responsive experiences as the model can adapt in real-time to user behavior. 
 
-  - Benefits
-  - Constraints
+However, On-Device Learning also comes with tradeoffs. The limited compute resources on consumer devices can make it challenging to run complex models locally. Datasets are also more restricted since they consist only of user-generated data from a single device. Additionally, updating models requires pushing out new versions rather than seamless cloud updates.
 
-## Continuous Learning
+On-Device Learning opens up new capabilities by enabling offline AI while maintaining user privacy. But it requires carefully managing model and data complexity within the constraints of consumer devices. Finding the right balance between localization and cloud offloading is key to delivering optimized on-device experiences.
 
-Explanation: Continuous learning is essential for embedded systems to adapt to new data and situations without requiring frequent updates from a central server.
+### Benefits
 
-  - Incremental Algorithms
-  - Adaptability
+#### Privacy and Data Security
 
-## Federated Machine Learning
+One of the significant advantages of on-device learning is the enhanced privacy and security of user data. For instance, consider a smartwatch that monitors sensitive health metrics such as heart rate and blood pressure. By processing data and adapting models directly on the device, the biometric data remains localized, circumventing the need to transmit raw data to cloud servers where it could be susceptible to breaches.
+
+Server breaches are far from rare, with millions of records compromised annually. For example, the 2017 Equifax breach exposed the personal data of 147 million people. By keeping data on the device, the risk of such exposures is drastically minimized. On-device learning acts as a safeguard against unauthorized access from various threats, including malicious actors, insider threats, and accidental exposure, by eliminating reliance on centralized cloud storage.
```

Comment:
The Equifax breach statement needs a reference since we are referring to concrete numbers.

Reply:
OK 👍 I will put that in